src/tests/user.test.ts

```diff
@@ -16,10 +16,78 @@ describe("User Tests", () => {
         email: faker.internet.email(),
         password: faker.internet.password(),
         name: faker.person.fullName()
-    }
+    };
+
     it("should register a User", async () => {
         const res = await request(app).post('/api/users/register').send(randomUser);
         expect(res.status).toBe(201);
         expect(res.body.message).toBe("User Registered Succefully");
     });
+
+    it("should not register a User with duplicate email", async () => {
+        const res = await request(app).post('/api/users/register').send(randomUser);
+        expect(res.status).toBe(409);
+        expect(res.body.message).toBe("Email is already taken");
+    });
+
+    it("should return 400 for a missing name", async () => {
+        const invalidUser = {
+            email: faker.internet.email(),
+            password: faker.internet.password(),
+            name:""
+        };
+        const res = await request(app).post('/api/users/register').send(invalidUser);
+        expect(res.status).toBe(400);
+        console.log(res.body);
+        expect(res.body.error).toContain('"name" is not allowed to be empty');
+    });
+
+    it("should return 400 for missing email", async () => {
+        const invalidUser = {
+            email: "",

```

Comment:
Suggestion: it will look better if you remove the email from the invalid user object